parsers/jscript.c

```diff
@@ -1029,7 +1030,30 @@ static void readTokenFullRaw (tokenInfo *const token, bool include_newlines, vSt
 		case ')': token->type = TOKEN_CLOSE_PAREN;			break;
 		case ';': token->type = TOKEN_SEMICOLON;			break;
 		case ',': token->type = TOKEN_COMMA;				break;
-		case '.': token->type = TOKEN_PERIOD;				break;
+		case '.':
+			{
+				token->type = TOKEN_PERIOD;
+
+				int d = getcFromInputFile ();
+				if (d != '.')
+				{
+					ungetcToInputFile (d);
+					break;
+				}
+
+				d = getcFromInputFile ();
+				if (d != '.')
+				{
+					ungetcToInputFile (d);
+					ungetcToInputFile ((int)'.');
+					break;
+				}
+
+				token->type = TOKEN_DOTS;
+				if (repr)
+					vStringCatS (repr, "..");
```

Comment:
Should this be 3 dots?

Reply:
The first dot is already appended to `repr` at line 1023:
```
  1018          if (repr && c != EOF)
  1019          {
  1020                  if (i > 1)
  1021                          vStringPut (repr, ' ');
  1022                  vStringPut (repr, c);
  1023          }
```

Reply:
I will add a comment explaining why "TWO dots" is enough here.

Reply:
See #3436.

Reply:
I added a comment in a newly added commit.